src/helpers.ts

```diff
@@ -34,21 +65,27 @@ export function delay(ms: number) {
 export function make_request_timing({
   request_start,
   response_start,
-  body_read_end = 0,
-  decompress_end = 0,
-  deserialize_end = 0,
-  requestCompressionDuration = 0,
-}: MakeRequestTiming): RequestTiming {
+  body_read_end,
+  decompress_end,
+  deserialize_end,
+  requestCompressionDuration,
+}: {
+  request_start: number;
+  response_start: number;
+  body_read_end?: number;
+  decompress_end?: number;
+  deserialize_end?: number;
+  requestCompressionDuration?: number;
+}): RequestTiming {
+  const deserialize_start = decompress_end ?? body_read_end;
   return {
     response_time: response_start - request_start,
     body_read_time: body_read_end ? body_read_end - response_start : 0,
     compress_time: requestCompressionDuration ? requestCompressionDuration : 0,
     decompress_time:
       decompress_end && body_read_end ? decompress_end - body_read_end : 0,
```

Comment:
still being reported as 0 instead of null

Reply:
for SemVer, not sure if this should be considered a breaking change and we update the MAJOR version or if it can be argued that this is a "fix" in a way

Reply:
arguably body_read_time could be 0 for head requests since we know we didn't spend any time reading it

my issue with decompress_time is that decompression could still happen within fetch() but we just don't know how long it took. so reporting 0 would be misleading